### DISCUSSION TITLE
NoDataBlockWarning: 1 blocks contained no data

### DISCUSSION THREAD

**Discussion by maucollu** [georgebv/pyextremes#81] [Q&A]
Hi,
I have a question that is more for a statistician than for a computer scientist, but I trust this community is full of people that are both!

I am deriving extreme values of significant wave height for a given location, having 30 years of hourly data.
I'm using block maxima, with a number of maxima per year going from 1 to 5.
No problem so far.

The issue is when I bin the data by direction, i.e., I would like to know what is the extreme value only considering the significant wave heights of the waves coming from one direction.
It is expected that, if in one direction there are not many data, there may be a possibility that, for a particular block, I do not have a data point, i.e., for 5 events per year, I do not have 5 sea state with mean wave direction from a direction in the bin.

Looking at the instructions, it seems that the way to deal with this is to set the `errors` parameter in the `model.get_extremes` to:
- `coerce`, i.e., get extreme values for blocks with no data as mean of all other extreme events in the series with index being the middle point of corresponding interval
- 'ignore', i.e. ignore blocks with no data coerce

It would seem to me that 'ignore' would be the best option, but would like to hear other opinions on this....

Many thanks

**Comment by georgebv** [comment]
Hi @maucollu . Have you tried fitting the model using both options (ignoring and coercing)? I wonder if there is going to be any difference in the results. Unless you are missing a lot for a particular direction that is - in that case you cannot get a meaningful result regardless. You could use variable binning to ensure you have enough data for each. Or you could even overlap your bins to get better coverage. The idea is that, if your approach is sound, your results should more-or-less converge for different options.
